### ISSUE TITLE
EarlyExit.UselessElseIf and yield

### ISSUE THREAD

**Issue by yethee** [slevomat/coding-standard#640]
Rule `SlevomatCodingStandard.ControlStructures.EarlyExit` changes logic, when elseif branch contains `yield` statement.

Example code:

```php
function example(int $value): iterable
{
    if (1 === $value) {
        yield 'one';
    } elseif (2 === $value) {
        yield 'two';
    } else {
        yield 'many';
    }
}
```

Report of phpcs:

```
--------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------
 9 | ERROR | [x] Use if instead of elseif.
   |       |     (SlevomatCodingStandard.ControlStructures.EarlyExit.UselessElseIf)
--------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------------------
```

After fix CS isssue:

```php
function example(int $value): iterable
{
    if (1 === $value) {
        yield 'one';
    }

    if (2 === $value) {
        yield 'two';
    } else {
        yield 'many';
    }
}
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/287ac3347c47918c0bf5e10335e36197ea10894c
